var/spack/repos/builtin/packages/hypre-cmake/package.py

```diff
@@ -90,38 +91,35 @@ def setup_build_environment(self, env):
 
     @run_after("install")
     def cache_test_sources(self):
+        if "+mpi" not in self.spec:
+            print("Package must be installed with +mpi to cache test sources")
+            return
+
```

Comment:
```suggestion
        # Customize the examples makefile before caching it
        makefile = join_path(self.extra_install_tests, "Makefile")
        filter_file(r"^HYPRE_DIR\s* =.*", f"HYPRE_DIR = {self.prefix}", makefile)
        filter_file(r"^CC\s*=.*", "CC = " + self.spec["mpi"].mpicc, makefile)
        filter_file(r"^F77\s*=.*", "F77 = " + self.spec["mpi"].mpif77, makefile)
        filter_file(r"^CXX\s*=.*", "CXX = " + self.spec["mpi"].mpicxx, makefile)
        filter_file(r"^LIBS\s*=.*", f"LIBS = -L$(HYPRE_DIR)/lib64 -lHYPRE -lm $(CUDA_LIBS) $(DOMP_LIBS)", makefile)

```

Reply:
This is the approach I've seen most and I'm not aware of an alternative at this time.